packages/frontend/core/src/blocksuite/ai/chat-panel/chat-panel-messages.ts

```diff
@@ -5,46 +5,33 @@ import {
   DocModeProvider,
   FeatureFlagService,
 } from '@blocksuite/affine/shared/services';
-import {
-  isInsidePageEditor,
-  type SpecBuilder,
-} from '@blocksuite/affine/shared/utils';
+import { type SpecBuilder } from '@blocksuite/affine/shared/utils';
 import type { BaseSelection } from '@blocksuite/affine/store';
-import {
-  AiIcon,
-  ArrowDownBigIcon as ArrowDownIcon,
-} from '@blocksuite/icons/lit';
+import { ArrowDownBigIcon as ArrowDownIcon } from '@blocksuite/icons/lit';
 import { css, html, nothing, type PropertyValues } from 'lit';
 import { property, query, state } from 'lit/decorators.js';
 import { repeat } from 'lit/directives/repeat.js';
 import { debounce } from 'lodash-es';
 
-import {
-  EdgelessEditorActions,
-  PageEditorActions,
-} from '../_common/chat-actions-handle';
 import { AffineIcon } from '../_common/icons';
-import {
-  type AIError,
-  PaymentRequiredError,
-  UnauthorizedError,
-} from '../components/ai-item/types';
-import { AIChatErrorRenderer } from '../messages/error';
+import { type AIError, UnauthorizedError } from '../components/ai-item/types';
 import { AIProvider } from '../provider';
 import {
   type ChatContextValue,
-  type ChatItem,
   type ChatMessage,
   isChatMessage,
 } from './chat-context';
+import { ChatPanelAssistantMessage } from './chat-panel-assistant-message';
+import { ChatPanelUserMessage } from './chat-panel-user-message';
 import { HISTORY_IMAGE_ACTIONS } from './const';
 import { AIPreloadConfig } from './preload-config';
 
-const AffineAvatarIcon = AiIcon({
-  width: '20px',
-  height: '20px',
-  style: 'color: var(--affine-primary-color)',
-});
+// 注册自定义元素
+customElements.define(
+  'chat-panel-assistant-message',
+  ChatPanelAssistantMessage
+);
+customElements.define('chat-panel-user-message', ChatPanelUserMessage);
 
```

Comment:
Wrap your custom element definitions into an effect fuction and call it in `registerAIEffects`.

```
function effects() {
customElements.define(
  'chat-panel-assistant-message',
  ChatPanelAssistantMessage
);
customElements.define('chat-panel-user-message', ChatPanelUserMessage);
}
```

Reply:
https://affine-pro.slack.com/archives/C05QXLZC1B4/p1726194095677309